src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -255,8 +276,7 @@ private synchronized void processCISShadow(String version, Map<String, Object> d
 
             try {
                 for (CertificateGenerator cg : monitoredCertificateGenerators) {
-                    cg.generateCertificate(connectivityInformation::getCachedHostAddresses,
-                            "connectivity info was updated");
+                    cg.generateCertificate(() -> cachedHostAddresses, "connectivity info was updated");
```

Comment:
why is this changing? calling the method should have the same effect right?

Reply:
should have the same effect, just being defensive if a future change ends up messing w/ cache concurrently or something

Reply:
esp since generateCertificate can take a while